benchcab/repository.py

```diff
@@ -119,18 +106,89 @@ def build(self, modules: list[str], verbose=False) -> None:
             )
         remove_module_lines(tmp_script_path)
 
-        args: list[str] = []
-        if internal.MPI and self.build_script is None:
-            args.append("mpi")
-
         with chdir(build_script_path.parent), self.modules_handler.load(
             modules, verbose=verbose
         ):
             self.subprocess_handler.run_cmd(
-                shlex.join([f"./{tmp_script_path.name}", *args]),
```

Comment:
Suggestion, but why not abstract the shlex wrapping in the subprocess handler itself, rather than having to manually wrap arguments each time it is called?

Consider changing the subprocess handler(?) to accept the script and subsequent arguments, applying the shlex wrapping and executing.

i.e. something along the lines of...

```shell
# UNTESTED!!!
def execute(script, *args):
   subprocess.run(shlex.join([script, args]))
```

...obviously following the execution logic you already having in place.

Reply:
I think I will leave out the shlex wrapping feature for now as this was actually the only place where we used `shlex.join()` and it has now been removed. But it might be an idea for the future since shell-escaped commands are currently not enforced by the subprocess handler.